### PR TITLE
Support edit journal

### LINKED ISSUE #40: support edit journal
add functionality for editing the journal

### LINKED ISSUE #44: [Journal] Implement the Delete Journal Functionality
Implement the feature which will be used to delete specific journals.

### PR DESCRIPTION
**Pull request by sanjithdevineni**
Adding Edit, Delete, Save buttons
changing the title of the journal on the canvas, when the title is inputted

### CODE REVIEW

File: journal/journal-app.js [hunk 1]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
+    journalWidgetTitle.classList.add('journal-widget-title')
+    journalWidgetTitle.textContent = 'Insert Title';
+    journalElementTest.append(journalWidgetTitle);
+    journalElementTest.setAttribute('widget-id', journalID);
+
+    // Create a container for the buttons
+    var buttonContainer = document.createElement('div');
+    buttonContainer.style.position = 'absolute';
+    buttonContainer.style.top = '5px';
+    buttonContainer.style.right = '5px';
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Why do we modify the styling of the element here instead of the CSS file?

File: journal/journal-app.js [hunk 2]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
+
+    // Create the 'Edit' button
+    var editButton = document.createElement('button');
+    editButton.innerText = 'Edit';
+    editButton.style.fontSize = '12px';
+    editButton.style.padding = '5px 10px';
+    editButton.style.marginRight = '5px';  // Space between buttons
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Why do we modify the styling of the element here instead of the CSS file?

File: journal/journal-app.js [hunk 3]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
+
+    // Create the 'Delete' button
+    var deleteButton = document.createElement('button');
+    deleteButton.innerText = 'Delete';
+    deleteButton.style.fontSize = '12px';
+    deleteButton.style.padding = '5px 10px';
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Why do we modify the styling of the element here instead of the CSS file?

File: journal/journal-app.js [hunk 4]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
+        const currentJournalID = widget.getAttribute('widget-id');
+
+        // Remove the journal element from the DOM
+        widget.remove();
+
+        // Remove the journal entry from localStorage
+        const journalList = getJournals();
+        const updatedJournalList = journalList.filter(journal => journal.id != currentJournalID);
+        saveNotes(updatedJournalList);
+    }
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Should this function be outside of the `createJournalWidget` function?

File: journal/journal-app.js [hunk 5]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
-            
-            journalContainer.append(journalElementTest);
-            
-            // Save new journal to localStorage
-            journalList.push(journalTemplate);
-            saveNotes(journalList);
+        const journalEntries = document.querySelectorAll('.journal-entry');
+        hideOtherJournalEntries(journalEntries, currentJournalID);
+        let modal = document.getElementById('modal');
+        modal.style.display = 'block';
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Should this function be outside of the `createJournalWidget` function?

File: journal/journal-app.js [hunk 6]
```diff
@@ -106,25 +157,56 @@ function createJournalElement(id, title, documentation, reflection, modalRef) {
...
+    saveButton.innerText = 'Save';
+
+    // Apply styles to the 'Save' button
+    saveButton.style.position = 'absolute';
+    saveButton.style.top = '5px';
+    saveButton.style.right = '5px';
+    saveButton.style.fontSize = '12px';
+    saveButton.style.padding = '5px 10px';
+
+    journalBody.style.position = 'relative';
```
Comment (kennethkietvuong) [inline journal/journal-app.js]: Again, why do we modify the styling here instead of the CSS file?

**Review by kennethkietvuong — CHANGES_REQUESTED** [review]
Overall, great work on the edit journal functionality. There are some concerns and questions I have from the review comments I have left, but here is a short summary:

- Why modify the styling of the element in the JS instead of the CSS file?
- There are two inner functions inside `createJournalWidget`, should those be outside?

I also locally tested your branch. From my testing:

- Edit/Delete/Save buttons work as intended
  - Editing title changes widget title
  - Delete button deletes widget from page and localStorage
  - Save button changes widget title
 - Refreshing or reloading the page **does not save content of the title, documentation, or reflection**
   - Saving the content to localStorage has not been set up yet, but we should further discuss this with the team as we have not integrated a means of SQLite at all as our database to presumably store the content?

File: journal/journal-app.js (hunk 1 above)
Comment (adironene) [reply journal/journal-app.js]: ^^ I think we should make a separate CSS file for the styling of journals

File: journal/journal-app.js [hunk 7]
```diff
@@ -32,69 +31,121 @@ function addJournalNew(journalContainer, existing) {
...
+        // Creating journal widget for new journal
+        createJournalWidget(journalContainer, journalID);
+
+        // Save new journal to localStorage
+        journalList.push(journalTemplate);
+        saveNotes(journalList);
+    }
+}
+
+// Extracted function to create and append the journal widget
```
Comment (adironene) [inline journal/journal-app.js:58]: We should start doing JSDocs for functions so it's easier for reviewers to understand types of param, output, and what function does

File: journal/journal-app.js (hunk 1 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved it to css file

File: journal/journal-app.js (hunk 2 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved to css file

File: journal/journal-app.js (hunk 3 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved to css file

File: journal/journal-app.js (hunk 4 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved it outside of the createJournalWidget function

File: journal/journal-app.js (hunk 5 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved it outside of the createJournalWidget function

File: journal/journal-app.js (hunk 6 above)
Comment (sanjithdevineni) [reply journal/journal-app.js]: moved this also to the css file

**Review by kennethkietvuong — APPROVED** [review]
Looks good!

**Review by adironene — APPROVED** [review]

**Comment by sanjithdevineni** [comment]
https://github.com/cse110-sp24-group30/cse110-sp24-group30/assets/103174323/9b632c55-e587-477e-bc5d-b8631163d48c

Video for UI functionality

**Pull request merged**